src/mapreduce.jl

```diff
@@ -60,12 +60,8 @@ end
 ## mapreduce ##
 ###############
 
-@inline function mapreduce(f, op, a::StaticArray, b::StaticArray...)
-    _mapreduce(f, op, same_size(a, b...), a, b...)
-end
-
-@inline function mapreduce(f, op, v0, a::StaticArray, b::StaticArray...)
-    _mapreduce(f, op, v0, same_size(a, b...), a, b...)
+@inline function mapreduce(f, op, a::StaticArray, b::StaticArray...; init = nothing)
```

Comment:
There's a trick to remove the `nothing` sentinel, used in `Base`.

Basically, use `; kw...` and we can see if the resulting NamedTuple has a field called `init` or not. A bit yucky but allows users to initiate a reduciton with `nothing` if they want.